src/web3/evm/triggers.ts

```diff
@@ -89,12 +90,14 @@ export class NewTransactionTrigger extends TriggerBase<{ chain: string | string[
     }
   }
 }
-export class NewEventTrigger extends TriggerBase<{
-  chain: string | string[];
-  contractAddress?: string;
-  eventDeclaration: string | string[];
-  parameterFilters: { [key: string]: unknown };
-}> {
+
+export class NewEventTrigger extends TriggerBase<NewEventInput, TriggerBasePayload, TriggerBaseState> {
+  public async executeProcessSignal(payload: TriggerBasePayload): Promise<boolean | void> {
+    if (!(await this.processSignal(payload))) {
+      return false;
+    }
+  }
```

Comment:
`processSignal` needs to be implemented by overriding the base function, not calling it from other function.

Reply:
@SAPikachu I tried this approach first by calling for example, in order to override `processSignal`:

```typescript
export async function safeDepositReceivedNative(input: ConnectorInput): Promise<TriggerBase> {
  const ret = new evm.NewEventTrigger(
    (await sanitizeParameters({
      ...input,
      fields: {
        ...(input.fields as object),
        eventDeclaration: "SafeReceived(address indexed sender, uint256 value)",
        parameterFilters: {},
      },
    })) as TriggerInit<NewEventInput, TriggerBasePayload, TriggerBaseState>
  );
  ret.processSignal = async (payload: TriggerBasePayload) => {
    if (String(payload.value) === "0") {
      return false;
    }
    payload.valueFormatted = ethers.utils.formatUnits(payload.value as string, "ether");
  };
  return ret;
}
``` 

But `processSignal` is marked as [protected](https://github.com/grindery-io/grindery-nexus-common-utils/blob/68c0bbaeedb56c0fed0baa3ae09dff5066b69495/src/connector/triggerBase.ts#L66C17-L66C17) in the `TriggerBase` class so that I don't think it can be access from exterior. I saw this was not the case previously with the `on` statement [here](https://github.com/grindery-io/grindery-nexus-common-utils/blob/75ccab3a9d89d0c9ac61732a2812b5a60a23f202/src/connector/triggerBase.ts#L88). Is there any reason why this is `protected` or maybe I missed something here?

Reply:
FYI `processSignal` is not supposed to be called from exterior, it will be called from within TriggerBase and we should create new subclass and override it if needed, like what I did in my commit.

Reply:
@SAPikachu Yes got that afterwards reading your commit :)